### DISCUSSION TITLE
Upgrading prop_dateString

### DISCUSSION THREAD

**Discussion by maccman** [xaviergonz/mobx-keystone#240] [Q&A]
I'm trying to upgrade mobx-keystone. It looks like some of the properties mappers have been removed. What do you recommend I use instead of `prop_dateString`?

**Comment by xaviergonz** [comment]
Yeah, sorry about that. Those transforms were making the typings slower/harder to reason about and they didn't work in many cases (e.g. a set of dates, etc.).

For now I recommend you to switch to a getter/setter:

```ts
class M extends Model({ timestamp: prop<number>() }) {
  get date() {
    return new Date(this.timestamp)
  }
  
  @modelAction
  setDate(date: Date) {
    this.timestamp = +date;
  }
}

const m = new M({
  timestamp: +(new Date())
})
```

I know it is typing a bit more, but it shouldn't be *that* bad

**Reply by maccman** [reply]
Is there any benefit to storing Dates as strings or as integers?

**Reply by xaviergonz** [reply]
it makes the dates serializable: JSON snapshot / patches compatibility

**Reply by xaviergonz** [reply]
Or if the question is if it is better to store dates as ISO strings or integers, personally I prefer integers, but it shouldn't matter.
